src/solvers/gradient_descent.jl

```diff
@@ -240,7 +240,15 @@ function gradient_descent!(
     ),
     stopping_criterion::StoppingCriterion=StopAfterIteration(200) |
                                           StopWhenGradientNormLess(1e-8),
-    debug=stepsize isa ConstantStepsize ? [DebugWarnIfCostIncreases()] : [],
+    debug=if (get_manopt_parameter(:Mode) == "Tutorial")
```

Comment:
This is so much logic that I would suggest making a separate function for computing this default.

Reply:
BTW, I'd also suggest making a helper `is_tutorial_mode` function.

Reply:
Yes, I am not yet sure we might have more modes, but if it basically remains this one, then that function sounds fair.

Reply:
Will add the easy access method tomorrow. It will make a few places more readable.